com.unity.visualeffectgraph/Editor/ShaderGraph/VFXSubTarget.cs

```diff
@@ -263,9 +263,15 @@ static PragmaCollection ApplyPragmaReplacement(PragmaCollection pragmas, VFXSRPB
                 foreach (var pragma in pragmas)
                 {
                     var currentPragma = pragma;
-                    var replacement = shaderGraphSRPInfo.pragmasReplacement.FirstOrDefault(o => o.oldDesc.value == pragma.descriptor.value);
-                    if (!string.IsNullOrEmpty(replacement.newDesc.value))
+                    var replacements = shaderGraphSRPInfo.pragmasReplacement.Where(o => o.oldDesc.value == pragma.descriptor.value);
```

Comment:
Why not FirstOrDefault directly?

Reply:
I mean what's the point of doing a Where + Any + First? It seems to interate on the whole enumerable always + allocate more to me?

Reply:
Indeed, it will iterate twice : I added a new behavior where we can remove a pragam if equals to `kPragmaDescriptorNone`.
The problem of the FirstOrDefault is the return value in case of fail, it's not `null` but `default(T)`, T is a structure and we can't really make the difference between `kPragmaDescriptorNone.value` &  `default(T).value`.

Reply:
Naïve fix https://github.com/Unity-Technologies/Graphics/pull/7025/commits/882bdb7476c05c179b12cffe3331ea531f60705a